### PR TITLE
Miscellaneous changes and refactoring

### PR DESCRIPTION
**Pull request by kjais1720**
## Overview
- Made Explore & Common user profile routes public. Now the user can view the explore page and checkout user profiles without logging in.
- Added edit comment feature. 
- Added new loader, `DotsLoader` 

## Preview
- https://hooked-social.vercel.app

### CODE REVIEW

File: src/components/CreateComment.jsx [hunk 1]
```diff
@@ -1,75 +1,131 @@
...
-import { postComment } from "services";
-import { useDispatch } from "react-redux";
+import { postComment, editComment } from "services";
+import { useSelector, useDispatch } from "react-redux";
 import { ProfileImage } from "./ProfileImage";
-import { useState } from "react";
-export function CreateComment({ postId }) {
-  const [formData, setFormData] = useState({
+import { Spinner } from "./Spinner";
+export const CreateComment = forwardRef(({ postId, commentToEdit }, ref) => {
```
Comment (t007rushi) [inline src/components/CreateComment.jsx:8]: clever use of forwardRef
nicely done @kjais1720

File: src/components/CreateComment.jsx [hunk 2]
```diff
@@ -1,75 +1,131 @@
...
               className="border-0 p-2 outline-0 dark:bg-dark-100 dark:text-gray-200"
             ></textarea>
-            <button
-              type="submit"
-              className="ml-auto rounded-2xl bg-dark-200 py-2 px-4 text-primary dark:bg-primary dark:text-dark-200"
-            >
-              Comment
-            </button>
-          </form>
+          </form> */}
```
Comment (t007rushi) [inline src/components/CreateComment.jsx]: remove comment if not needed/forgot

File: src/slices/userSlice.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ export const userSlice = createSlice({
     builder
       //Login/Signup
       .addCase(authenticateUser.pending, (state) => {
```
Comment (t007rushi) [inline src/slices/userSlice.js:49]: isn't builder outdated in RTK
can directly write cases ````[authenticateUser.pending]``` format

**Review by t007rushi — APPROVED** [review]
Look for suggestions else good to go ahead as always

**Review by ganesh-kanchi — APPROVED** [review]
The code is very well written and the folder structure is well maintained. this can be merged.

File: src/components/CreateComment.jsx (hunk 1 above)
Comment (kjais1720) [reply src/components/CreateComment.jsx:8]: Thank you 🤗

File: src/slices/userSlice.js (hunk 3 above)
Comment (kjais1720) [reply src/slices/userSlice.js:49]: Yes, but at the time, I was just following the official docs and it used builder, will refactor it later

**Pull request merged**